### DISCUSSION TITLE
Do not use tile.openstreetmap.org

### DISCUSSION THREAD

**Discussion by tnibler** [immich-app/immich#3866] [General]
OSM tile servers are primarily meant for map editors and are funded by donations to the OSM foundation.
They are not for use in production applications, as described in their [usage policy](https://operations.osmfoundation.org/policies/tiles/).

As this project keeps growing (quickly), more and more users will be hitting the OSM tile servers which is not a nice thing to do :( 

Immich should use a different source for its map tiles, and at least implement some caching to reduce requests to the tile servers as soon as possible (I couldn't find any caching at first glance at least).

**Comment by lefuturiste** [comment]
Hi! Thanks for rising this issue (tbc I'm not an Immich core contributor). But I also think this is an issue.
I think we should as a first step allow to easily change the tile URL to use another provider in the admin settings.
Providing a complete OSM tile rendering stack along with immich is possible but difficult and will increase resource usage significantly, may be we can ship along something less heavy with less OSM details (like a reduced quality version of OSM data, with only the majors map elements).

**Comment by tnibler** [comment]
I haven't looked too deeply into it yet, but https://protomaps.com/ could be a fairly good fit for this application. The API is more permissive for non commercial use, and you could even host the files on a central server only for Immich, and then add caching on top of that so every Immich server keeps its local copy of the map data it needs. That should be fairly cost effective I think.

**Reply by bo0tzz** [reply]
Protomaps looks pretty awesome, and it might even be feasibile to host tiles ourselves using that.

**Comment by bo0tzz** [comment]
Is there any caching we can add on top of the default browser caching? And are there other providers than the OSM tile servers that we could use? I agree that we should not use a public service against their terms like this. 

If there is no alternative public service, what are the other options? Would we need to ship a tileserver with Immich, or run our own somewhere under immich.app?

@alextran1502 to make up for this a little bit, what do you think about making a [donation to OSM](https://supporting.openstreetmap.org/) using some of the funds that people have donated to Immich?

**Comment by tnibler** [comment]
So, any updates about this? It's really not cool to abuse a free community resource like that.

**Reply by bo0tzz** [reply]
I'm entirely in agreement that we should change this, I'm just not sure how to move forwards. If you have answers to the questions I posed above, that would help.

**Comment by lefuturiste** [comment]
Well, I think I will be working on a MR to :
- allow to configure the tile URL to allow to change provider or style
And I'm in favor to change the default URL to protomap or something similar.

**Reply by tnibler** [reply]
Ok nice, thank you! I just believe that in the age of Cloudflare and WEI, it's all the more important to show that free and open services can still exist on the web, and to respect the rules they set out of courtesy.

**Reply by bo0tzz** [reply]
Does protomaps have a public/free endpoint that we could use (and would that be OK with them)?

**Reply by tnibler** [reply]
It does have a free API, you have to sign up for a key though (maybe the creator is open to arrange something with Immich?). But also the `pmtiles` [basemap file](https://protomaps.com/docs/downloads/basemaps) is only 100GB, which could be hosted in a bucket just for Immich pretty cheaply.

**Comment by bo0tzz** [comment]
We've reached out to the creator of Protomaps to explore that option.

**Comment by bo0tzz** [comment]
Update: While protomaps isn't an option (unless we set it up ourselves), we do now have an alternative tileserver to use. However, this involves changing to a vector tile format which means we need to move the map implementation we use from leaflet to maplibre, which will take a bit more time.

**Reply by daniele-athome** [reply]
What if e.g. I host assets from protomaps somewhere only for myself? Would that work?

**Reply by bo0tzz** [reply]
Leaflet requires raster tiles, whereas protomaps only supports vector tiles, so at the moment that won't work no. Once we have maplibre swapped in it should be possible to use either.

**Reply by daniele-athome** [reply]
In the meatime I've created an account on [Maptiler](https://www.maptiler.com/cloud/) and have been using it on my Immich instance. It's very easy to configure and their free plan has a fair amount of requests/month - especially if it's just few people using the Immich instance. Configure the tile endpoint like this:

```
https://api.maptiler.com/maps/streets-v2/{z}/{x}/{y}.png?key=YOUR_API_KEY
```

**Comment by bo0tzz** [comment]
#4294 will move Immich to using Maplibre with another default tileserver.

**Comment by bo0tzz** [comment]
Closed by #4294
